packages/lib/date-ranges.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, expect, it } from "vitest";
+
+import dayjs from "@calcom/dayjs";
+
+import { buildDateRanges, processDateOverride, processWorkingHours } from "./date-ranges";
+
+describe("processWorkingHours", () => {
+  it("should return the correct working hours given a specific availability, timezone, and date range", () => {
+    const item = {
+      days: [1, 2, 3, 4, 5], // Monday to Friday
+      startTime: new Date(Date.UTC(2023, 5, 12, 8, 0)), // 8 AM
+      endTime: new Date(Date.UTC(2023, 5, 12, 17, 0)), // 5 PM
+    };
+
+    const timeZone = "America/New_York";
+    const dateFrom = dayjs.utc().startOf("day").day(2).add(1, "week");
+    const dateTo = dayjs.utc().endOf("day").day(3).add(1, "week");
+
+    const results = processWorkingHours({ item, timeZone, dateFrom, dateTo });
+
+    expect(results.length).toBe(2); // There should be two working days between the range
+    // "America/New_York" day shifts -1, so we need to add a day to correct this shift.
+    expect(results[0]).toEqual({
```

Comment:
Can we use [toMatchInlineSnapshot](https://vitest.dev/api/expect.html#tomatchinlinesnapshot) here instead of computing the answer as well?